src/rgw/rgw_datalog.h

```diff
@@ -219,7 +226,8 @@ class RGWDataChangesLog {
   int start(const RGWZone* _zone, const RGWZoneParams& zoneparams,
 	    RGWSI_Cls *cls_svc, librados::Rados* lr);
 
-  int add_entry(const RGWBucketInfo& bucket_info, int shard_id);
+  int add_entry(const RGWBucketInfo& bucket_info, int shard_id,
+		const rgw::bucket_log_layout_generation& gen);
```

Comment:
can you please put the shard_id argument after log layout? the layout depends on the bucket, and the shard id depends on the layout. the order of arguments should reflect that

Reply:
Done